util-cache/src/main/scala/sbt/util/CacheImplicits.scala

```diff
@@ -8,7 +8,47 @@
 
 package sbt.util
 
+import com.github.benmanes.caffeine.cache.{ Cache as CCache, Caffeine }
+import java.nio.file.{ Files, NoSuchFileException }
+import java.nio.file.attribute.BasicFileAttributes
 import sjsonnew.BasicJsonProtocol
+import xsbti.{ HashedVirtualFileRef, PathBasedFile }
 
 object CacheImplicits extends CacheImplicits
-trait CacheImplicits extends BasicCacheImplicits with BasicJsonProtocol
+trait CacheImplicits extends BasicCacheImplicits with BasicJsonProtocol:
+
+  private val stampCache: CCache[String, (String, Long, Long)] = Caffeine
+    .newBuilder()
+    .maximumSize(50000)
```

Comment:
Should this size be configurable? I could see repositories of a sufficient size blowing through this limit.

Obviously it's a tradeoff, but I figure people will want to own the tradeoff with memory consumption.

Reply:
I can try to make this configurable via a global setting.